### PR TITLE
CheckedDiv/CheckedRem + saturating family for HeaplessBigInt

### PR DESCRIPTION
**Pull request by kaidokert**
Next in the parity series. Knocks out **two of the three** remaining `num_traits::PrimInt` blockers (the third is `Num`/`from_str_radix`, coming with the string surface).

**CheckedDiv / CheckedRem** (`div_rem.rs`, Nct — division is Nct-only): return `None` on divide-by-zero instead of the `div_rem_impl` assert, so div-by-zero is finally reachable *without* going through the panicking operator. Inherent `checked_div`/`checked_rem` + const_num_traits + num_traits trait forms.

**Saturating** (`arith.rs`): inherent `saturating_add`/`sub` (P-generic) + `saturating_mul`, const_num_traits `Saturating{Add,Sub,Mul}` (Nct, matching the existing `Checked*` gating), and `num_traits::Saturating` (Nct, needed by PrimInt).

**Correctness note worth a look:** the saturation target is the all-ones value at the **operand** width `max(a.len, b.len)` (via `max_at_len`), *not* `Bounded::max_value()`. FixedUInt uses `Bounded::max_value()` because its width *is* `N`; on heapless `Bounded::max_value()` is the **CAP-wide** max, which would over-saturate a sub-capacity value (e.g. two 32-bit values in a 256-bit carrier must saturate to the 32-bit max, not the 256-bit max). There's a dedicated heapless-internal test for exactly this — the kind of case the fixed-width carrier harness can't reach (its backings have width == CAP).

**Tests:** generic `for_both_carriers` bodies in `carrier_generic.rs` (const traits) and `carrier_num_traits.rs` (num_traits bridges), plus the sub-capacity saturation-width and div-by-zero heapless-internal tests. Verified default / nightly / --no-default-features; clippy clean.

## Summary by Sourcery

Add checked division/remainder and saturating arithmetic support to HeaplessBigInt, aligning its behavior and trait coverage with the fixed-width carrier and num_traits requirements.

New Features:
- Provide inherent checked_div and checked_rem methods on HeaplessBigInt that return None on divide-by-zero.
- Add inherent saturating_add, saturating_sub, and saturating_mul operations on HeaplessBigInt that saturate at the operand width rather than the carrier capacity.
- Expose const_num_traits SaturatingAdd, SaturatingSub, and SaturatingMul implementations for HeaplessBigInt, plus num_traits::Saturating for Nct.
- Implement const_num_traits CheckedDiv and CheckedRem, and num_traits::CheckedDiv/CheckedRem, for HeaplessBigInt in the Nct personality.

Enhancements:
- Introduce an internal max_at_len helper to compute the all-ones value at a given limb width for correct saturation semantics on heapless carriers.

Tests:
- Extend generic carrier tests to cover saturating arithmetic and checked division/remainder across both carriers.
- Add heapless-internal tests to verify operand-width saturation behavior and that checked_div/checked_rem return None on divide-by-zero.
- Add num_traits bridge tests validating Saturating and CheckedDiv/CheckedRem behavior for both carriers.

### CODE REVIEW

File: src/heapless/arith.rs [hunk 1]
```diff
@@ -207,6 +226,25 @@ impl<T: MachineWord, const CAP: usize, P: Personality> HeaplessBigInt<T, CAP, P>
         let (res, borrow) = self.overflowing_sub(other);
         if borrow { None } else { Some(res) }
     }
+
+    /// Saturating addition: on overflow, the all-ones value at the operands'
+    /// width `max(a.len, b.len)` (not the CAP-wide max), matching
+    /// `FixedUInt<T, width>::saturating_add`.
+    pub fn saturating_add(&self, other: &Self) -> Self {
+        let (res, overflow) = self.overflowing_add(other);
+        if overflow { max_at_len(res.len) } else { res }
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/heapless/arith.rs]: Good catch. Fixed in c8a5e3d by moving the inherent saturating_add/sub/mul to Nct-only impl blocks — the Ct personality no longer exposes them, matching the already-Nct-gated Saturating* trait forms. A branchless Ct saturating (like FixedUInt's const_ct_select) is left as future work, consistent with heapless keeping its Ct trait surface minimal; for now the leak surface is removed rather than made branchless.

File: src/heapless/arith.rs [hunk 2]
```diff
@@ -26,6 +26,25 @@ use core::marker::PhantomData;
...
+    let mut limbs = [zero::<T>(); CAP];
+    for l in &mut limbs[..len as usize] {
+        *l = <T as Bounded>::max_value();
+    }
+    HeaplessBigInt {
+        limbs,
+        len,
+        _p: PhantomData,
+    }
+}
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/heapless/arith.rs:46]: @gemini-code-assist The current code already does this — `for l in &mut limbs[..len as usize]` slices to the logical length up front and iterates it, so there's no per-element indexing or mid-loop bounds check; the suggested form is equivalent. Leaving as-is.
